app/services/structure_updater.rb

```diff
@@ -100,15 +100,26 @@ def from_csv
     end
 
     contains = files_by_fileset.keys.map do |sequence|
-      # Find the existing fileset
-      existing_fileset = model.structural.contains[sequence.to_i - 1]
+      fileset = fileset_for(sequence, files_by_fileset[sequence].first.label)
       attributes = fileset_attributes[sequence]
                    .merge(structural: { contains: files_by_fileset[sequence] })
-      existing_fileset.new(**attributes)
+      fileset.new(**attributes)
     end
     Success(model.structural.new(contains:))
   end
 
+  FILESET_NAMESPACE = 'https://cocina.sul.stanford.edu/fileset/'
+
+  # @param [int] sequence is the index of the fileset in the import
+  # @param [string] label the label to inject for a new fileset
+  def fileset_for(sequence, label)
+    model.structural.contains[sequence.to_i - 1].presence ||
```

Comment:
We've already asserted this is an Integer, why cast `.to_i`?

Reply:
Bad copypasta, the `.to_i` is meant to be in the call to make the assertion here true. Fixed. Thanks for catching that.